cf/commands/servicekey/delete_service_key_test.go

```diff
@@ -0,0 +1,148 @@
+		})
+
+		It("requires two arguments and one option to run", func() {
+			Expect(callDeleteServiceKey([]string{})).To(BeFalse())
+			Expect(callDeleteServiceKey([]string{"fake-arg-one"})).To(BeFalse())
+			Expect(callDeleteServiceKey([]string{"fake-arg-one", "fake-arg-two", "fake-arg-three"})).To(BeFalse())
+		})
+
+		It("fails when service instance is not found", func() {
+			requirementsFactory = &testreq.FakeReqFactory{LoginSuccess: true, ServiceInstanceNotFound: true}
+			Expect(callDeleteServiceKey([]string{"non-exist-service-instance"})).To(BeFalse())
+		})
+
+		It("fails when space is not targetted", func() {
+			requirementsFactory = &testreq.FakeReqFactory{LoginSuccess: true, TargetedSpaceSuccess: false}
+			Expect(callDeleteServiceKey([]string{"fake-service-instance", "fake-service-key"})).To(BeFalse())
+		})
+	})
+
+	Describe("requirements are satisfied", func() {
+		Context("deletes service key successfully", func() {
+			BeforeEach(func() {
+				serviceKeyRepo.GetServiceKeyMethod.ServiceKey = models.ServiceKey{
+					Fields: models.ServiceKeyFields{
+						Name:                "fake-service-key",
+						Guid:                "fake-service-key-guid",
+						Url:                 "fake-service-key-url",
+						ServiceInstanceGuid: "fake-service-instance-guid",
+						ServiceInstanceUrl:  "fake-service-instance-url",
+					},
+					Credentials: map[string]interface{}{
+						"username": "fake-username",
+						"password": "fake-password",
+						"host":     "fake-host",
+						"port":     "3306",
+						"database": "fake-db-name",
+						"uri":      "mysql://fake-user:fake-password@fake-host:3306/fake-db-name",
+					},
+				}
+			})
+
+			It("deletes service key successfully when '-f' option is provided", func() {
+				requirementsFactory = &testreq.FakeReqFactory{LoginSuccess: true, ServiceInstanceNotFound: false, TargetedSpaceSuccess: true}
+
+				Expect(callDeleteServiceKey([]string{"fake-service-instance", "fake-service-key", "-f"})).To(BeTrue())
+				Expect(ui.Outputs).To(ContainSubstrings(
+					[]string{"Deleting key", "fake-service-key", "for service instance", "fake-service-instance", "as", "my-user"},
+					[]string{"OK"}))
+			})
+
+			It("deletes service key successfully when '-f' option is not provided and confirmed 'yes'", func() {
+				requirementsFactory = &testreq.FakeReqFactory{LoginSuccess: true, ServiceInstanceNotFound: false, TargetedSpaceSuccess: true}
+				ui.Inputs = append(ui.Inputs, "yes")
+
+				Expect(callDeleteServiceKey([]string{"fake-service-instance", "fake-service-key"})).To(BeTrue())
+				Expect(ui.Prompts).To(ContainSubstrings([]string{"Really delete the service key", "fake-service-key"}))
+				Expect(ui.Outputs).To(ContainSubstrings(
+					[]string{"Deleting key", "fake-service-key", "for service instance", "fake-service-instance", "as", "my-user"},
+					[]string{"OK"}))
+			})
+
+			It("deletes service key successfully when '-f' option is not provided and confirmed 'no'", func() {
```

Comment:
So this reads odd, I expected that it DOES NOT delete service key if -f option not provided and confirmed 'no'... otherwise, this is the same test as above. I think you may need to change the `It("...")` description string...